### DISCUSSION TITLE
New feature: sort songs by characteristics

### DISCUSSION THREAD

**Discussion by l-bar22** [SpotX-Official/SpotX#313] [Ideas]
Hey,I don't know if anyone has thought about it,but it would be cool to be able to sort the songs in the playlist by their characteristics.for example,if you are listening to a playlist that has dance songs and at the same time slow songs,there are times when the latter are a bit choppy or vice versa.if you could sort them by their characteristics you would have more or less of the same style and it would be another way to sort them.the code could be something like this.the code could be something like this.

`import spotipy
from spotipy.oauth2 import SpotifyOAuth

sp = spotipy.Spotify(auth_manager=SpotifyOAuth(client_id="<YOUR_SPOTIPY_CLIENT_ID>", client_secret="<YOUR_SPOTIPY_CLIENT_SECRET>", redirect_uri="<YOUR_REDIRECT_URI>", scope=["user-library-read", "playlist-read-private"]))

def sort_playlist_by_feature(playlist_id, feature):
    playlist = sp.playlist(playlist_id)
    tracks = playlist["tracks"]["items"]
    track_ids = [t["track"]["id"] for t in tracks]
    audio_features = sp.audio_features(track_ids)
    sorted_tracks = [t for _, t in sorted(zip(audio_features, tracks), key=lambda x: x[0][feature])]
    return sorted_tracks

Example usage: sort the playlist with ID "7kUJ0vUzX9UxlZH8jKUjGv" by danceability
sorted_tracks = sort_playlist_by_feature("7kUJ0vUzX9UxlZH8jKUjGv", "danceability")
`

**Comment by amd64fox** [comment]
It's not that difficult to write code to sort by track characteristics or some other parameters, the question here is how you are going to embed your code in the native Spotify client.
